pennylane/ops/qubit/hamiltonian.py

```diff
@@ -219,6 +219,27 @@ def label(self, decimals=None, base_label=None, cache=None):
         decimals = None if (len(self.parameters) > 3) else decimals
         return super().label(decimals=decimals, base_label=base_label or "𝓗", cache=cache)
 
+    @property
+    def data(self):
+        """Return the operator data
+
+        Returns:
+            List[float]: coefficients in the Hamiltonian expression as a list
+        """
+        return self._data
+
+    @data.setter
+    def data(self, data):
+        """Set the operator data, and update the coefficients accordingly.
+
+        Raises:
+        ValueError: if the number of coefficients does not match the number of observables
+        """
+        if qml.math.shape(data)[0] != qml.math.shape(self.ops)[0]:
+            raise ValueError("The number of coefficients and operators does not match.")
+        self._data = list(data)
+        self._coeffs = qml.math.stack(data) if data else []
```

Comment:
So are `data` and `coeffs the same, except for a `qml.math.stack` call, and the fact that setting `data` updates `coeffs` but not vice versa (and `coeffs` indeed can not be set)? 
A little independently of this PR, this seems a bit redundant?